packages/server/src/controllers/roadmaps/filter.ts

```diff
@@ -1,25 +1,76 @@
+import { z } from "zod";
+import { Request, Response } from "express";
 import database from "../../database";
-
-// utils
 import logger from "../../utils/logger";
 import error from "../../errorResponse.json";
 
-export async function filter(_, res) {
+const querySchema = z.object({
+  first: z.coerce.number().min(1).max(20).default(20),
+  after: z.string().uuid().optional()
```

Comment:
@kris70lesgo The TS is logging error about `ZodString' is deprecated`. Can you check it?